src/test/java/org/kie/processmigration/service/PlanServiceImplTest.java

```diff
@@ -16,52 +16,106 @@
+
+    @AfterEach
+    @Transactional
+    void cleanUp() {
+        Plan.deleteAll();
+    }
 
     @Test
-    public void testSaveAndFindAll() {
-        // Given
-        assertNotNull(planService);
+    void testService() {
+        assertThat(planService, CoreMatchers.notNullValue());
+        assertThat(planService.findAll(), empty());
+    }
 
-        Plan plan = new Plan();
-        plan.setName("name");
-        plan.setSource(new ProcessRef().setContainerId("containerId").setProcessId("sourceProcessId"));
-        plan.setTarget(new ProcessRef().setContainerId("targetContainerId").setProcessId("targetProcessId"));
-        plan.setDescription("description");
+    @Test
+    void testCreateAndFindAll() {
+        // Given
+        Plan plan = new Plan()
+                .setName("name")
+                .setSource(new ProcessRef()
+                        .setContainerId("containerId").setProcessId("sourceProcessId"))
+                .setTarget(new ProcessRef().setContainerId("targetContainerId").setProcessId("targetProcessId"))
+                .setDescription("description");
 
         // When
-        getEntityManager().getTransaction().begin();
-        planService.create(plan);
-        getEntityManager().getTransaction().commit();
+        Plan result = planService.create(plan);
 
         // Then
         List<Plan> plans = planService.findAll();
 
-        assertNotNull(plans);
-        assertEquals(1, plans.size());
+        assertThat(plans, notNullValue());
+        assertThat(plans, hasSize(1));
+        assertThat(plans.get(0), equalTo(result));
+    }
+
+    @Test
+    void testDelete() throws PlanNotFoundException {
+        assertThrows(PlanNotFoundException.class, () -> planService.delete(1L));
+        // Given
+        Plan plan = new Plan()
+                .setName("name")
+                .setSource(new ProcessRef()
+                        .setContainerId("containerId").setProcessId("sourceProcessId"))
+                .setTarget(new ProcessRef().setContainerId("targetContainerId").setProcessId("targetProcessId"))
+                .setDescription("description");
+        Plan result = planService.create(plan);
+
+        // When
+        assertThat(result, notNullValue());
+        assertThat(result.getId(), notNullValue());
+        assertThat(planService.delete(result.getId()), equalTo(plan));
+
+        // Then
+        assertThat(planService.findAll(), empty());
+    }
+
+
+    @Test
+    void testUpdate() throws PlanNotFoundException {
+        assertThrows(PlanNotFoundException.class, () -> planService.delete(1L));
+        // Given
+        Plan plan = new Plan()
+                .setName("name")
+                .setSource(new ProcessRef()
+                        .setContainerId("containerId").setProcessId("sourceProcessId"))
+                .setTarget(new ProcessRef().setContainerId("targetContainerId").setProcessId("targetProcessId"))
+                .setDescription("description");
```

Comment:
Maybe you can refactor and create a small method to create the Plan object, but just a suggestion, not needed